src/dict.h

```diff
@@ -80,8 +84,9 @@ typedef struct dictType {
      * enables one more optimization: to store a key without an allocated
      * dictEntry. */
     unsigned int keys_are_odd : 1;
-    /* TODO: Add a 'keys_are_even' flag and use a similar optimization if that
-     * flag is set. */
+    /* If embedded_entry flag is set, it indicates that a copy of the key is created and the key is embedded
+     * as part of the dict entry. */
+    unsigned int embedded_entry : 1;
```

Comment:
Correct. Added regular assert. I don't think it will be a lot of penalty. If you think otherwise, need to get in the helper debug assert methods to dict.